src/client/app/components/IntervalControlsComponent.tsx

```diff
@@ -0,0 +1,169 @@
+		setShowCustomDuration(isCustom);
+		setDaysCustom(duration.asDays());
+		setDays(isCustom ? CUSTOM_INPUT : duration.asDays().toString());
+	}, [duration]);
+
+	// Returns true if this is a valid duration.
+	const daysValid = (days: number) => {
+		return Number.isInteger(days) && days >= MIN_DAYS && days <= MAX_DAYS;
+	};
+
+	// Updates values when the standard duration menu for bar or map is used.
+	const handleDaysChange = (value: string) => {
+		if (value === CUSTOM_INPUT) {
+			// Set menu value from standard value to special value to show custom
+			// and show the custom input area.
+			setDays(CUSTOM_INPUT);
+			setShowCustomDuration(true);
+		} else {
+			// Set the standard menu value, hide the custom duration input
+			// and duration for graphing.
+			// Since controlled values know it is a valid integer.
+			setShowCustomDuration(false);
+			updateDurationChange(Number(value));
+		}
+	};
+
+	// Updates value when the custom duration input is used for bar or map.
+	const handleCustomDaysChange = (value: number) => {
+		setDaysCustom(value);
+	};
+
+	const handleEnter = (key: string) => {
+		// This detects the enter key and then uses the previously entered custom
+		// duration to set the duration for the graphic.
+		if (key === 'Enter') {
+			updateDurationChange(daysCustom);
+		}
+	};
+
+	const updateDurationChange = (value: number) => {
+		// Update if okay value. May not be okay if this came from user entry in custom form.
+		if (daysValid(value)) {
+			dispatch(graphSlice.actions.updateDuration(moment.duration(value, 'days')));
+		}
+	};
+
+	// Define the initial set of options for the dropdown when the chart type is not 'compare'.
+	const options = [
+		{ label: 'day', value: '1' },
+		{ label: 'week', value: '7' },
+		{ label: '4.weeks', value: '28' }
+	];
+
+	// If the chart type is 'compare', we change the options to reflect comparison periods.
+	if (chartType === ChartTypes.compare) {
+		options.length = 0; // Clear the previous options
+		Object.entries(ComparePeriod).forEach(([key, value]) => {
+			options.push({ label: key, value: value.toString() });
+		});
+	}
+
+	return (
+		<div>
+			<div style={divTopBottomPadding}>
+				<p style={labelStyle}>
+					{(chartType === ChartTypes.bar && translate('bar.interval')) ||
+						(chartType === ChartTypes.map && translate('map.interval')) ||
+						(chartType === ChartTypes.compare && translate('compare.period'))}:
+					<TooltipMarkerComponent page='home' helpTextId={
+						chartType === ChartTypes.bar ? 'help.home.bar.days.tip' :
+							chartType === ChartTypes.map ? 'help.home.map.days.tip' :
+								'help.home.compare.period.tip'
+					} />
+				</p>
+				<Input
+					id='durationDays'
+					name='durationDays'
+					type='select'
+					value={chartType === ChartTypes.compare ? comparePeriod?.toString() : days}
+					onChange={e => handleDaysChange(e.target.value)}
+				>
+					{options.map(option => (
+						<option value={option.value} key={option.label}>
```

Comment:
For four weeks, the old version had "4 Weeks" on the menu (see https://openenergydashboard.org/helpV1_0_0/compareGraphic/). I think that is better and matches the other ones. I'm not sure the best way but maybe treat the items as keys and translate via data.ts. Also, these options for compare are not translating so that would address that too.